lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
This is probably better pushed down into the event sink as @andrewgr once suggested instead of duplicating this check across aggregates.

You will also have the added benefit of hooking into it when emitting events from reactors I believe.